tests/failpoints/cases/test_disk_full.rs

```diff
@@ -0,0 +1,209 @@
+    pd_client.must_add_peer(region.get_id(), new_peer(old_leader.get_store_id(), 20000));
+    let region = cluster.get_region(&key);
+    // for i in region.get_peers() {
+    //     println!("peer info: {}-{}", i.get_store_id(), i.get_id());
+    // }
+    assert!(region.get_peers().len() == 3);
+    //failed to propose confchange, error: Other("[components/raftstore/src/store/peer.rs:2469]: [region 1000] 1002 unsafe to perform conf change [peer { id: 2000 store_id: 1 }], before: voters: 1004 voters: 1002 voters: 1003, after: voters: 2000 voters: 1002 voters: 1004 voters: 1003, truncated index 5, promoted commit index 0")
+    fail::remove(DISK_FULL_LEADER);
+}
+
+fn fail_follower_full(cluster: &mut Cluster<ServerCluster>) {
+    std::thread::sleep(Duration::from_secs(2));
+    let key = String::from("2222").into_bytes();
+    let value = String::from("22220000").into_bytes();
+    let region = cluster.get_region(&key);
+    let target_peer = region
+        .get_peers()
+        .iter()
+        .find(|x| x.get_store_id() == 3)
+        .unwrap();
+    let leader = cluster.leader_of_region(region.get_id()).unwrap();
+    if leader.get_store_id() != target_peer.get_store_id() {
+        cluster.must_transfer_leader(region.get_id(), (*target_peer).clone());
+    }
+    let leader = cluster.leader_of_region(region.get_id()).unwrap();
+    let region = cluster.get_region(&key);
+    let follower1 = region
+        .get_peers()
+        .iter()
+        .find(|x| x.get_store_id() == 1)
+        .unwrap();
+    let follower2 = region
+        .get_peers()
+        .iter()
+        .find(|x| x.get_store_id() == 2)
+        .unwrap();
+
+    fail::cfg(DISK_FULL_FOLLOWER_1, "return").unwrap();
+    assert!(leader.get_store_id() == 3);
+    assert!(follower1.get_store_id() == 1);
+    assert!(follower2.get_store_id() == 2);
+    {
+        cluster.must_put(&key, &value);
+        let leader_state = cluster.raft_local_state(region.get_id(), leader.get_store_id());
+        let follower_state_1 = cluster.raft_local_state(region.get_id(), follower1.get_store_id());
+        let follower_state_2 = cluster.raft_local_state(region.get_id(), follower2.get_store_id());
+        assert!(leader_state.get_last_index() == follower_state_2.get_last_index());
+        assert!(leader_state.get_last_index() != follower_state_1.get_last_index());
+    }
+    {
+        //cluster.must_transfer_leader(region.get_id(), (*follower1).clone());
+        let timer = Instant::now();
+        loop {
+            cluster.reset_leader_of_region(region.get_id());
+            let cur_leader = cluster.leader_of_region(region.get_id());
+            if let Some(ref cur_leader) = cur_leader {
+                assert!(
+                    !(cur_leader.get_id() == follower1.get_id()
+                        && cur_leader.get_store_id() == follower1.get_store_id())
+                );
+            }
+            if timer.elapsed() > Duration::from_secs(2) {
```

Comment:
So this part will keep 2 seconds? It's too long. We can write code like this:
```
cluster.transfer_leader(target);
cluster.async_command_on_peer(Put, ...);
assert(put_success);
```

Reply:
Maybe not good, because we hope the transfer failed, and I've not found good way to check it.